File: backend/UI-MVC/Models/Dto/IdeaDto.cs
using Conversey.BL.Domain.Common;
using Conversey.BL.Domain.Ideation;

namespace Conversey.UI_MVC.Models.Dto;

public class IdeaDto
{
    public int Id { get; set; }
    public string Content { get; set; }
    public Slug ProjectId { get; set; }
    public int TopicId { get; set; }
    public Guid YouthId { get; set; }
    public DateTime SubmissionDate { get; set; }
    public ModerationStatus Status { get; set; }
    public IEnumerable<ReactionDto> Reactions { get; set; }

    public static IdeaDto From(Idea idea)
    {
        return new IdeaDto
        {
            Id = idea.Id,
            Content = idea.Content,
            ProjectId = idea.Project.Id,
            TopicId = idea.Topic.Id,
            YouthId = idea.Youth.Id,
            SubmissionDate = idea.SubmissionDate,
            Status = idea.Status,
            Reactions = (idea.Reactions ?? Array.Empty<IdeaReaction>())
                .GroupBy(r => r.Emoji)
                .Select(g => new ReactionDto { Emoji = g.Key, Count = g.Count() })
        };
    }
}

File: backend/UI-MVC/Assets/api/dtos/ideaDto.ts
type ApiSlugValue = string | { text?: string; Text?: string }

export interface ApiIdeaTopicDto {
    id?: number
    Id?: number
    projectId?: number | ApiSlugValue
    ProjectId?: number | ApiSlugValue
    title?: string
    Title?: string
    prompt?: string
    Prompt?: string
    order?: number
    Order?: number
}

export interface ApiIdeaDto {
    id?: number
    Id?: number
    projectId?: number | ApiSlugValue
    ProjectId?: number | ApiSlugValue
    topicId?: number
    TopicId?: number
    content?: string
    Content?: string
    status?: string | number
    Status?: string | number
    youthId?: string
    YouthId?: string
    submissionDate?: string
    SubmissionDate?: string
    qualityNudgeBypassed?: boolean
    QualityNudgeBypassed?: boolean
    semanticCategories?: string[]
    SemanticCategories?: string[]
    reactions?: Array<{ emoji?: string; Emoji?: string; count?: number; Count?: number }>
    Reactions?: Array<{ emoji?: string; Emoji?: string; count?: number; Count?: number }>
}

export interface ApiSubmitIdeaRequestDto {
    content: string
    youthId: string
    qualityNudgeBypassed?: boolean
}
